### PR TITLE
EDGSSRS-1: Create the edge-search-srs module

### PR DESCRIPTION
**Pull request by IlliaDaliek**
Jira: https://issues.folio.org/browse/EDGSSRS-1

### PURPOSE
1) Create and set up a spring-based "edge-search-srs" module. 
2) Integrate the module with the Jenkins pipeline.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -2,7 +2,7 @@ buildMvn {
   publishModDescriptor = 'yes'
   publishAPI = 'no'
   mvnDeploy = 'yes'
-  runLintRamlCop = 'yes'
+  runLintRamlCop = 'no'
```
Comment (dcrossleyau) [inline Jenkinsfile]: The new api-lint (https://dev.folio.org/guides/api-lint/) is the replacement. Please add it when your src is ready.

File: README.md [hunk 2]
```diff
@@ -1 +1,23 @@
-# edge-search-srs
\ No newline at end of file
+# edge-search-srs
+
+Copyright (C) 2018-2019 The Open Library Foundation
+
```
Comment (dcrossleyau) [inline README.md:4]: The date-range in the top of the README should just be "2021", as explained in https://dev.folio.org/guidelines/create-new-repo/

File: README.md [hunk 3]
```diff
@@ -1 +1,23 @@
...
+This software is distributed under the terms of the Apache License,
+Version 2.0. See the file "[LICENSE](LICENSE)" for more information.
+
+## Introduction
+
+Microservice to allow 3rd party discovery services to perform FOLIO patron
+actions via the discovery service's UI.
+
```
Comment (dcrossleyau) [inline README.md:12]: The README "Introduction" section looks like a copy of mod-patron. Is that correct?

File: descriptors/ModuleDescriptor-template.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
+{
+  "id": "${artifactId}-${version}",
+  "name": "Edge search srs API",
```
Comment (dcrossleyau) [inline descriptors/ModuleDescriptor-template.json:3]: Being a spring-based module, i gather that you need to fix maven replacement tokens in ModuleDescriptor [FOLSPRINGB-4](https://issues.folio.org/browse/FOLSPRINGB-4)

File: descriptors/ModuleDescriptor-template.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  "provides": [],
+  "requires": [
+    {
+      "id": "login",
+      "version": "5.0 6.0 7.0"
+    }
+  ],
+  "permissionSets": [],
+  "launchDescriptor": {
+    "dockerImage": "${artifactId}:${version}",
```
Comment (dcrossleyau) [inline descriptors/ModuleDescriptor-template.json]: Being a spring-based module, i gather that you need to fix maven replacement tokens in ModuleDescriptor [FOLSPRINGB-4](https://issues.folio.org/browse/FOLSPRINGB-4)

**Review by dcrossleyau — CHANGES_REQUESTED** [review]
Additional comments:

As per https://dev.folio.org/guidelines/create-new-repo/
the required personal data disclosure form is missing.

There is no GitHub Team access enabled.

File: README.md (hunk 2 above)
Comment (IlliaDaliek) [reply README.md:4]: > Additional comments:
> 
> As per https://dev.folio.org/guidelines/create-new-repo/
> the required personal data disclosure form is missing.
> 
> There is no GitHub Team access enabled.

Hi @dcrossleyau, thank you for so helpful notes. I fixed all of them. Could you please take a final look again to ensure that I did everything correctly?) Thank you

File: Jenkinsfile (hunk 1 above)
Comment (IlliaDaliek) [reply Jenkinsfile]: Hi @dcrossleyau , thank you for so helpful notes. I fixed all of them. Could you please take a final look again to ensure that I did everything correctly?) Thank you

**Comment by IlliaDaliek** [comment]
> Additional comments:
> 
> As per https://dev.folio.org/guidelines/create-new-repo/
> the required personal data disclosure form is missing.
> 
> There is no GitHub Team access enabled.

Hi @dcrossleyau, thank you for so helpful notes. I fixed all of them. Could you please take a final look again to ensure that I did everything correctly?) Thank you

File: descriptors/ModuleDescriptor-template.json (hunk 5 above)
Comment (dcrossleyau) [reply descriptors/ModuleDescriptor-template.json]: This is still not fixed.

File: README.md (hunk 2 above)
Comment (dcrossleyau) [reply README.md:4]: This is a new module. So:
Copyright (C) 2021 The Open Library Foundation

**Review by dcrossleyau — APPROVED** [review]
Still no Team access configured for this repository. That can come later.

There are a couple of remaining items noted above.

**Review by andrii-novytskyi — APPROVED** [review]

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+    <groupId>org.springframework.boot</groupId>
+    <artifactId>spring-boot-starter-parent</artifactId>
+    <version>2.3.4.RELEASE</version>
+    <relativePath/>
+  </parent>
+
+  <groupId>org.folio</groupId>
+  <artifactId>edge-search-srs</artifactId>
+  <version>1.0-SNAPSHOT</version>
+
```
Comment (dcrossleyau) [inline pom.xml:17]: This version number is broken, should be x.y.z-SNAPSHOT

Is there some Spring Way template that has a broken POM? We saw this same error with mod-bursar-export [FOLIO-3001](https://issues.folio.org/browse/FOLIO-3001)

**Review by dcrossleyau — CHANGES_REQUESTED** [review]

**Review by oleksandr-dekin — APPROVED** [review]

File: pom.xml (hunk 6 above)
Comment (TotoroDevEbsco) [reply pom.xml:17]: This version is specified when creating a project. Fixed to 1.0.0-SNAPSHOT

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,12 @@
+FROM folioci/alpine-jre-openjdk11:latest
+
+ENV VERTICLE_FILE edge-search-srs-1.0.0-SNAPSHOT.jar
+
```
Comment (dcrossleyau) [inline Dockerfile:4]: Why is the version number hard-coded? That seems prone to future failure.

**Review by dcrossleyau — APPROVED** [review]
I am doing the "Approve" because you have fixed the topic of my previous review -- the POM version number.

However see the Dockerfile.

It seems to me that Spring Way projects need to address general common issues before proliferation.

I do not know why DevOps people are being involved in pull-request reviews. Especially delays from the other side of the planet.

It is worrying that other developers are approving, when there are problems.

File: Dockerfile (hunk 7 above)
Comment (IlliaDaliek) [reply Dockerfile:4]: I used it from other spring based projects.

Comment (craigmcnally) [reply Dockerfile:4]: I agree this is bad practice.

**Comment by craigmcnally** [comment]
@IlliaDaliek Edge modules should leverage the edge-common framework

**Pull request merged**